spec/models/mixins/purging_mixin_spec.rb

```diff
@@ -1,6 +1,19 @@
 RSpec.describe PurgingMixin do
   let(:example_class) { PolicyEvent }
   let(:purge_date) { 2.weeks.ago }
+  purge_by_delete_classes, purge_by_destroy_classes = ActiveRecord::Base.descendants.select { |m| m.ancestors.include?(PurgingMixin) && m.base_model == m }.partition { |m| m.purge_method == :delete }
```

Comment:
I had to add the base_model check as we don't need to test all the descendant classes such as:

```
  ManageIQ::Providers::Azure::ContainerManager::ContainerGroup.purge_method is destroy
  ManageIQ::Providers::Kubernetes::ContainerManager::ContainerGroup.purge_method is destroy
  ManageIQ::Providers::Vmware::ContainerManager::ContainerGroup.purge_method is destroy
  ManageIQ::Providers::OracleCloud::ContainerManager::Container.purge_method is destroy
```

Reply:
TODO: add base_model? check and use it here